src/chocolatey/infrastructure.app/commands/ChocolateyExportCommand.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright © 2017 - 2018 Chocolatey Software, Inc
+// Copyright © 2011 - 2017 RealDimensions Software, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace chocolatey.infrastructure.app.commands
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Text;
+    using System.Xml;
+    using attributes;
+    using commandline;
+    using configuration;
+    using filesystem;
+    using infrastructure.commands;
+    using logging;
+    using services;
+    using tolerance;
+
+    [CommandFor("export", "exports list of currently installed packages")]
+    public class ChocolateyExportCommand : ICommand
+    {
+        private readonly INugetService _nugetService;
+        private readonly IFileSystem _fileSystem;
+
+        public ChocolateyExportCommand(INugetService nugetService, IFileSystem fileSystem)
+        {
+            _nugetService = nugetService;
+            _fileSystem = fileSystem;
+        }
+
+        public void configure_argument_parser(OptionSet optionSet, ChocolateyConfiguration configuration)
+        {
+            optionSet
+                .Add("o=|output-file-path=",

```

Comment:
Do we use `o` anywhere else with regards to output?

Additionally we use `output` in a few places, let's keep this consistent.

Reply:
![](https://media3.giphy.com/media/Xw6yFn7frR3Y4/giphy.gif)

Reply:
@ferventcoder no, this is the first place that `o` is introduced as a short option, so I think we are good to use it, if you agree.

Reply:
Does this need to be an option switch? To me reading it, it feels kind of like i'd expect `choco export ./packages.config` to work without requiring an `-o`. The filepath here feels like it should be the primary argument?

Reply:
That is a good point. In order to be in-keeping with the rest of the Chocolatey commands, I have added the ability to either provide the explicit option, or not.  So the following all work:

```
    choco export
    choco export "'c:\temp\packages.config'"
    choco export -o="'c:\temp\packages.config'"
    choco export --output-file-path="'c:\temp\packages.config'"
```